rflx/specification/parser.py

```diff
@@ -1191,17 +1189,18 @@ def __init__(self, skip_verification: bool = False, cached: bool = False) -> Non
         self.__cache = Cache(cached)
 
     def __convert_unit(
-        self, spec: Specification, specfile: Path = None, transitions: List[ID] = None
+        self, spec: Specification, filename: Path = STDIN, transitions: List[ID] = None
     ) -> RecordFluxError:
         transitions = transitions or []
         error = RecordFluxError()
 
         if spec:
-            parent = specfile.parent if specfile else Path(".")
-            filename = parent / Path(f"{spec.f_package_declaration.f_identifier.text.lower()}.rflx")
-            self.__specifications[filename] = (specfile, spec)
+            packagefile = filename.parent / Path(
+                f"{spec.f_package_declaration.f_identifier.text.lower()}.rflx"
+            )
+            self.__specifications[packagefile] = (filename, spec)
```

Comment:
What is the idea of prefixing the `packagefile` with `filename.parent`? This isn't done in the old parser. It doesn't prevent name conflicts, as there could be still two files with the same package identifier in the same directory.

Reply:
It's unnecessary and changed in 3ef392f9

Reply:
Errors are undetected for two files with the same name in different directories:

```Console
% cat a/test.rflx
package Test is

end Test;
% cat b/test.rflx
package Test is

   type Boolean is (A, B) with Size => 2;

end Test;
% rflx check a/test.rflx b/test.rflx
Parsing a/test.rflx
Processing Test
% rflx check b/test.rflx            
Parsing b/test.rflx
Processing Test
b/test.rflx:3:4: model: error: illegal redefinition of built-in type "Boolean"
b/test.rflx:3:9: model: error: name conflict for type "__BUILTINS__::Boolean"
__BUILTINS__:0:0: model: info: previous occurrence of "__BUILTINS__::Boolean"
```

Reply:
As we'd need to redo the fix for this issue when addressing #572, we'll leave this issue as is. I created a new issue #589 which will be merged in the same PR as #572.